crates/ty_python_semantic/src/semantic_index/use_def.rs

```diff
@@ -639,6 +680,12 @@ impl<'db> Iterator for DeclarationsIterator<'_, 'db> {
 
 impl std::iter::FusedIterator for DeclarationsIterator<'_, '_> {}
 
+#[derive(Debug, PartialEq, Eq, salsa::Update)]
+struct ReachableDefinitions {
+    bindings: Bindings,
+    declarations: Declarations,
```

Comment:
It's not totally clear to me that reusing `Declarations` and `Bindings` structs, adding `PreviousDefinitions` flag when recording a definition, and leaving their constraint-tracking functionality unused in this case, is the clearest approach here? How much code would actually be duplicated if we used bespoke simpler vectors of bindings/declarations here?

Really fine with either option here, just curious if it's a tradeoff you considered.